client/src/controllers/LocaleController.test.js

```diff
@@ -0,0 +1,137 @@
+      <option value="Asia/Katmandu">Asia/Katmandu</option>
+      <option value="Atlantic/Canary">Atlantic/Canary</option>
+      <option value="Australia/South">Australia/South</option>
+      <option value="Brazil/East">Brazil/East</option>
+      <option value="Canada/Atlantic">Canada/Atlantic</option>
+      <option value="Chile/Continental">Chile/Continental</option>
+      <option value="EST">EST</option>
+      <option value="Etc/GMT-7">Etc/GMT-7</option>
+      <option value="Europe/Brussels">Europe/Brussels</option>
+      <option value="GMT">GMT</option>
+      <option value="Indian/Maldives">Indian/Maldives</option>
+      <option value="Pacific/Tarawa">Pacific/Tarawa</option>
+      <option value="UTC">UTC</option>
+      <option value="Universal">Universal</option>
+      <option value="Zulu">Zulu</option>
+    `;
+
+    app = Application.start();
+    app.register('w-locale', LocaleController);
+    app.register('w-init', InitController);
+
+    await Promise.resolve();
+  };
+
+  afterEach(() => {
+    app?.stop();
+    jest.clearAllMocks();
+  });
+
+  describe('localizing time zone options', () => {
+    it('should append localized time zone labels to the options', async () => {
+      document.documentElement.lang = 'en-US';
+      await setup(/* html */ `
+        <select
+          name="locale-current_time_zone"
+          data-controller="w-init w-locale"
+          data-action="w-init:ready->w-locale#localizeTimeZoneOptions"
+          data-w-locale-server-time-zone-param="Europe/London"
+        >
+        </select>
+      `);
+
+      expect(select.getAttribute('data-controller')).toEqual('w-locale');
+      const selected = select.selectedOptions[0];
+      expect(selected).toBeTruthy();
+      expect(selected.value).toEqual('');
+      expect(selected.textContent).toEqual(
+        'Use server time zone: GMT (Greenwich Mean Time)',
+      );
+      expect(select).toMatchSnapshot();
+    });
+  });
+
+  it('should localize to the current HTML locale and use the server time zone param for the default', async () => {
+    document.documentElement.lang = 'id-ID';
+    await setup(/* html */ `
+      <select
+        name="locale-current_time_zone"
+        data-controller="w-init w-locale"
+        data-action="w-init:ready->w-locale#localizeTimeZoneOptions"
+        data-w-locale-server-time-zone-param="Asia/Jakarta"
+      >
+      </select>
+    `);
+
+    expect(select.getAttribute('data-controller')).toEqual('w-locale');
+    const selected = select.selectedOptions[0];
+    expect(selected).toBeTruthy();
+    expect(selected.value).toEqual('');
+    expect(selected.textContent).toEqual(
+      'Use server time zone: WIB (Waktu Indonesia Barat)',
+    );
+    expect(select).toMatchSnapshot();
+  });
+
+  it('should skip updating the default option if server time zone is not provided', async () => {
+    document.documentElement.lang = 'ar';
+    await setup(/* html */ `
```

Comment:
@laymonage Just curious, you used the `/* html */` convention in your tests. Is this something specific to your IDE?

Do you think we should adopt this or something similar in all our unit tests?

```
    await setup(/* html */ `
```

I have thought about a unit test `html` template literal function that could maybe be used by IDEs but also help validate the HTML for testing.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

I also saw that we had some malformed HTML in some other unit tests, maybe we could help improve the state of this in our Jest testing.

Reply:
Yeah, I use the following extension to add syntax highlighting with comments: https://github.com/mjbvz/vscode-comment-tagged-templates

I did consider adding a template literal function, but I couldn't be bothered making a separate PR for it. I'd give a +1 on that, as from my experience that'd also allow the HTML inside to be formatted by Prettier if you have the [Lit extension](https://github.com/runem/lit-analyzer/tree/master/packages/vscode-lit-plugin) installed. Though, as you said, it should only be used in tests (as such a function could incur a tiny runtime overhead).